q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,311 @@
+        # checks that all the items in the expected value list are strings.
+        str_items = [x for x in rule_value if isinstance(x, str)]
+        if len(str_items) == len(rule_value):
+            return True
+    return False
+
+
+def check_ontology(rule_value):
+    """Check that the user-defined value for the
+    rule "ontology" is correctly formatted.
+
+    ontology must be a str
+        Name of the ontology to lookup within for control vocabulary, e.g.
+                    ontology: Gazetteer
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "ontology".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    if isinstance(rule_value, str):
+        if rule_value in ['Gazetteer ontology']:
+            return True
+    return False
+
+
+def check_remap(rule_value):
+    """Check that the user-defined value for the
+    rule "remap" is correctly formatted.
+
+    remap must be a dict
+        Apply the rule of remapping variables values: replace
+        all instances of a given value by another one value.
+                    remap:
+                      str: str
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "remap".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    if isinstance(rule_value, dict):
+        str_values = [
+            x for x, y in rule_value.items() if
+                isinstance(x, (str, int, float)) and
+                isinstance(y, (str, int, float))
+        ]
+        if len(str_values) == len(rule_value.keys()):
+            return True
+    return False
+
+
+def check_validation(rule_value):
+    """Check that the user-defined value for the
+    rule "validation" is correctly formatted.
+
+    validation must be a dict
+        Apply the rule of validating variables values: replace by the blank
+        (or the missing) value every sample entry that for the current variable has
+        in one or multiple other variable(s) a null or missing value for that sample.
+                    validation:
+                      force_to_blank_if:
+                        is null:
+                        - variable
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "validation".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    # accepted structure for the "validation" rule
+    possible_validations = {
+        'force_to_blank_if': {'is null'}
+    }
+    # check the first level is a dict
+    if isinstance(rule_value, dict):
+        # get the current keys
+        rule_value_set = set(rule_value)
+        # get the accepted keys from the current keys
+        accepted_set = rule_value_set & set(possible_validations)
+        # if there is not more keys than the accepted ones
+        if not len(rule_value_set ^ accepted_set):
```

Comment:
If I understand correctly, you are checking that `rule_value_set` is a subset of `accepted_set` or for that matter of `rule_value_set` right? Suggest using the issubset method.

Reply:
right and I forgot to add a check for must-be-numerics:

```
def check_normalization(rule_value):
    """Check that the user-defined value for the rule
    "normalization" is a dictionary which keys are no
    other than "minimum", "maximum" or "gated_value",
    and which values are numeric.

    Parameters
    ----------
    rule_value
        Parsed rule value for the rule "normalization".

    Returns
    -------
    bool
        True if successful, False otherwise.
    """
    allowed = {'maximum', 'minimum', 'gated_value'}
    rule_value_set = set(rule_value)
    if isinstance(rule_value, dict) and rule_value_set.issubset(allowed):
        if len(rule_value) == sum(map(lambda x: isinstance(x, (int, float)), rule_value.values())):
            return True
    return False
```
that's sweet, thanks!